src/prefect/futures.py

```diff
@@ -612,19 +613,40 @@ def resolve_futures_to_states(
 
     Unsupported object types will be returned without modification.
     """
-    futures: set[PrefectFuture[R]] = set()
+    return _resolve_futures(
+        expr,
+        resolve_fn=lambda future: future.state if future.wait() or True else None,
+    )
```

Comment:
I think this would be easier to read if the `resolve_fn` was declared as a separate function like if is in `resolve_futures_to_results`.
```suggestion
    def _resolve_state(future: PrefectFuture[R]):
        future.wait()
        return future.state
        
    return _resolve_futures(
        expr,
        resolve_fn=_resolve_state,
    )
```

Reply:
Thanks for catching this. I had done this originally and then asked an LLM to refactor my code. Completely missed this absolutely unhinged suggestion...